RecoEgamma/EgammaElectronProducers/src/LowPtGsfElectronFeatures.cc

```diff
@@ -164,6 +164,400 @@ namespace lowptgsfeleid {
+                    sc_clus2_deta = deta;
+                    sc_clus2_dphi = dphi;
+                  }
+                }
+                clusNum++;
+              }
+            }
+          } catch (...) {
+            std::cout << "caught an exception" << std::endl;
+          }
+        }
+      }
+    }  // clusters
+
+    // Out-of-range
+    if (eid_rho < 0)
+      eid_rho = 0;
+    if (eid_rho > 100)
+      eid_rho = 100;
+    if (eid_sc_eta < -5)
+      eid_sc_eta = -5;
+    if (eid_sc_eta > 5)
+      eid_sc_eta = 5;
+    if (eid_shape_full5x5_r9 < 0)
+      eid_shape_full5x5_r9 = 0;
+    if (eid_shape_full5x5_r9 > 2)
+      eid_shape_full5x5_r9 = 2;
+    if (eid_sc_etaWidth < 0)
+      eid_sc_etaWidth = 0;
+    if (eid_sc_etaWidth > 3.14)
+      eid_sc_etaWidth = 3.14;
+    if (eid_sc_phiWidth < 0)
+      eid_sc_phiWidth = 0;
+    if (eid_sc_phiWidth > 3.14)
+      eid_sc_phiWidth = 3.14;
+    if (eid_shape_full5x5_HoverE < 0)
+      eid_shape_full5x5_HoverE = 0;
+    if (eid_shape_full5x5_HoverE > 50)
+      eid_shape_full5x5_HoverE = 50;
+    if (eid_trk_nhits < -1)
+      eid_trk_nhits = -1;
+    if (eid_trk_nhits > 50)
+      eid_trk_nhits = 50;
+    if (eid_trk_chi2red < -1)
+      eid_trk_chi2red = -1;
+    if (eid_trk_chi2red > 50)
+      eid_trk_chi2red = 50;
+    if (eid_gsf_chi2red < -1)
+      eid_gsf_chi2red = -1;
+    if (eid_gsf_chi2red > 100)
+      eid_gsf_chi2red = 100;
+    if (eid_brem_frac < 0)
+      eid_brem_frac = -1;
+    if (eid_brem_frac > 1)
+      eid_brem_frac = 1;
+    if (eid_gsf_nhits < -1)
+      eid_gsf_nhits = -1;
+    if (eid_gsf_nhits > 50)
+      eid_gsf_nhits = 50;
+    if (eid_match_SC_EoverP < 0)
+      eid_match_SC_EoverP = 0;
+    if (eid_match_SC_EoverP > 100)
+      eid_match_SC_EoverP = 100;
+    if (eid_match_eclu_EoverP < -0.001)
+      eid_match_eclu_EoverP = -0.001;
+    if (eid_match_eclu_EoverP > 0.001)
+      eid_match_eclu_EoverP = 0.001;
+    eid_match_eclu_EoverP = eid_match_eclu_EoverP * 1.E7;
+    if (eid_match_SC_dEta < -10)
+      eid_match_SC_dEta = -10;
+    if (eid_match_SC_dEta > 10)
+      eid_match_SC_dEta = 10;
+    if (eid_match_SC_dPhi < -3.14)
+      eid_match_SC_dPhi = -3.14;
+    if (eid_match_SC_dPhi > 3.14)
+      eid_match_SC_dPhi = 3.14;
+    if (eid_match_seed_dEta < -10)
+      eid_match_seed_dEta = -10;
+    if (eid_match_seed_dEta > 10)
+      eid_match_seed_dEta = 10;
+    if (eid_sc_E < 0)
+      eid_sc_E = 0;
+    if (eid_sc_E > 1000)
+      eid_sc_E = 1000;
+    if (eid_trk_p < -1)
+      eid_trk_p = -1;
+    if (eid_trk_p > 1000)
+      eid_trk_p = 1000;
+    if (gsf_mode_p < 0)
+      gsf_mode_p = 0;
+    if (gsf_mode_p > 1000)
+      gsf_mode_p = 1000;
+    if (core_shFracHits < 0)
+      core_shFracHits = 0;
+    if (core_shFracHits > 1)
+      core_shFracHits = 1;
+    if (gsf_bdtout1 < -20)
+      gsf_bdtout1 = -20;
+    if (gsf_bdtout1 > 20)
+      gsf_bdtout1 = 20;
+    if (gsf_dr < 0)
+      gsf_dr = 5;
```

Comment:
is this intentionally set to 5? unlike other cases this is not a simple clamp/truncation, more like wrap-around

Reply:
I think so. @crovelli, please can you reply?

Reply:
dR<0 means that the track is not found. So we set it to the maximum allowed value, i.e. 5. So yes, it' intentional @cavalari